test/common/performance/aggregator/aggregator.go

```diff
@@ -276,19 +270,29 @@ func eventsToTimestampsArray(events *map[string]*timestamp.Timestamp) []time.Tim
 		t, _ := ptypes.Timestamp(v)
 		values = append(values, t)
 	}
-	sort.Slice(values, func(x, y int) bool { return values[x].Before(values[y]) })
 	return values
 }
 
 func publishThpt(timestamps []time.Time, q *quickstore.Quickstore, metricName string) error {
-	for i, t := range timestamps[1:] {
-		var thpt uint
-		j := i - 1
-		for j >= 0 && t.Sub(timestamps[j]) <= time.Second {
-			thpt++
-			j--
+	if len(timestamps) >= 2 {
+		sort.Slice(timestamps, func(x, y int) bool { return timestamps[x].Before(timestamps[y]) })
+		for i, t := range timestamps[1:] {
+			var thpt uint
+			j := i - 1
+			for j >= 0 && t.Sub(timestamps[j]) <= time.Second {
+				thpt++
+				j--
+			}
+			if qerr := q.AddSamplePoint(mako.XTime(t), map[string]float64{metricName: float64(thpt)}); qerr != nil {
+				return qerr
+			}
+		}
+	} else if len(timestamps) == 1 {
+		if qerr := q.AddSamplePoint(mako.XTime(timestamps[0]), map[string]float64{metricName: 1}); qerr != nil {
+			return qerr
 		}
-		if qerr := q.AddSamplePoint(mako.XTime(t), map[string]float64{metricName: float64(thpt)}); qerr != nil {
+	} else {
+		if qerr := q.AddSamplePoint(mako.XTime(time.Now()), map[string]float64{metricName: 0}); qerr != nil {
```

Comment:
Are you sure `time.Now()` doesn't lead to wrong graphs too because it increases the bench time to when the values are published? What if you use the first or last timestamp of whatever other data point (like the first or last sent event data point) as a "fallback" case when no errors are registered?

Reply:
I'm sure this change will fix the aggregate chart.

The run chart will still have problems, but it always has as what we've discussed in https://github.com/knative/eventing/pull/2218.